portal/auth.py

```diff
@@ -76,7 +76,7 @@ def wrapped_view(**kwargs):
 def login_role(view):
     @functools.wraps(view)
     def wrapped_view(**kwargs):
-        if g.user['role'] != 'teacher':
-            return redirect(url_for('portal.student'))
+        if g.user == None or g.user['role'] != 'teacher':
```

Comment:
you're definitely right for including a None option, but if g.user is none, they should be redirected to the login? maybe an error like "you must login to have access to the portal". I didnt think to include none, good call.